### ISSUE TITLE
[Suggestion] App switcher

### ISSUE THREAD

**Issue by System64fumo** [horriblename/hyprgrass#160]
I would love to see something like an app switcher like the ones found on mobile oses.
### Example
<a href="url"><img src="https://github.com/user-attachments/assets/e767097e-5d85-4e00-ad32-dca70e91f32b" align="left" width="400" ></a><br><br><br>

### Desired behavior
 * Swipe up (From the bottom edge) to enter the switcher, Cannot enter switcher unless there are 2 or more workspaces.
 * Tap on a workspace to exit the switcher.
 * Ability to scroll/slide between workspaces.
 * Swipe up on a toplevel to close it
 * Swipe down on a workspace to close every toplevel in it (Optional)

### Differences from hyprexpo
Hyprexpo has a static grid, This should be dynamic & scrollable.
Hyprexpo shows the wallpaper in every workspace, This should only show the toplevels in a workspace.

### Final thoughts
I understand that all of this might be too much to ask for and that you might not have the time nor motivation to work on such big of a task.
I'd happily donate if it motivates you or helps in any way.

**Comment by horriblename** [comment]
definitely out of scope for this project, but I'll think about it as a separate plugin

**Comment by System64fumo** [comment]
Awesome!
Please keep me updated :call_me_hand:

**Comment by horriblename** [comment]
[hyprspace](https://github.com/KZDKM/Hyprspace) might interest you

**Comment by horriblename** [comment]
can you elaborate? I haven't tried it but it does seem to have some basic touch support

**Comment by System64fumo** [comment]
I've tried it but couldn't get it working with touch yet (known issue)
The switcher can be configured almost the way that i want
As for the touch gesture support
I would love for it to integrate with hyprgrass's edge swipe feature

In other words
I'd like it to behave like mobile OS-es do
Swipe up to trigger the switcher
Swipe left or right to switch workspaces

**Comment by horriblename** [comment]
I've put up a fix for touch screen crashing

- switching workspace with hyprgrass gestures should just work in hyprspace
- you can toggle hyprspace with their dispatcher, hence:

   ```
   # untested
   hyprgrass-bind = , edge:d:u, overview:toggle 
   ```

**Comment by System64fumo** [comment]
Awesome work!

Regarding the dispatcher though..
I was kind of hoping to avoid doing that
I was looking for something seemless (Like with a trackpad for instance)

**Comment by horriblename** [comment]
well it's not _impossible_... I just don't know if it's a good idea. I'll cook something up

**Comment by System64fumo** [comment]
Likely a terrible idea

There are like 2 ways of going about doing this
1: Support this behavior from hyprspace (Copy bottom gesture code from hyprgrass to hyprspace)
2: Support this behavior from hyprgrass (Copy workspace preview code from hyprspace to hyprgrass)

**Comment by smathles** [comment]
Keen to see where this goes. The lack of touchscreen functionality with hyprspace is a bit painful atm.

I get it though if it's not exactly the intended scope of hyprgrass though lol, wrong plugin.

**Comment by System64fumo** [comment]
Yeah hope either the desired functionality gets added to hyprspace or horriblename starts their own plugin (Or adds support for this to hyprgrass)

**Comment by horriblename** [comment]
waiting for #174

TL;DR hyprgrass will be exposing hooks so that other plugins can have hyprgrass integration

**Comment by horriblename** [comment]
with #189 you can now emulate touchpad swipes. Not exactly the integration I promised, but it works pretty nicely with hyprspace.

the event hooks I talked about is currently being re-thought

**Comment by System64fumo** [comment]
Whoops forgot to reply to this, Amazing work!

Bit offtopic but i've noticed you're looking for on screen keyboards?
Could i interest you with my own?
https://github.com/System64fumo/sysboard

**Comment by horriblename** [comment]
Experimental hooks just merged #196, there's a (pretty bad) plugin that controls pulse-audio volume via edge gestures

waiting for [rendering fixes](https://github.com/KZDKM/Hyprspace/issues/131) for hyprspace integration and further testing

**Comment by System64fumo** [comment]
Woo hoo!
Good work!!
